Core/RoverControllerPresets/DriveControllers/DirectDriveController.cs

```diff
@@ -14,21 +17,45 @@ public RoverControl CalculateMoveVector()
 			Convert.ToSingle(LocalSettings.Singleton.Joystick.Deadzone)
 		);
 
-		Vector2 vec = new(
-			Input.GetAxis("rover_move_backward", "rover_move_forward"),
-			Input.GetAxis("rover_move_right", "rover_move_left")
-		);
-
-		if (LocalSettings.Singleton.SpeedLimiter.Enabled)
-			vec.X *= LocalSettings.Singleton.SpeedLimiter.MaxSpeed;
-
-		vec.X = Mathf.IsEqualApprox(vec.X, 0f, joyDeadZone) ? 0 : vec.X;
+		Vector3 vec;
+
+		switch (Mode)
+		{
+			case KinematicMode.Ackermann:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					0);
+				break;
+			case KinematicMode.Crab:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					Input.GetAxis("rover_move_down", "rover_move_up")
+				);
+				break;
+			case KinematicMode.Spinner:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					0f,
+					0f
+				);
+				break;
+			case KinematicMode.EBrake:
+				vec = new(0, 0, 0); //todo wheels in X pattern?
+				break;
+			default:
+				vec = new(0, 0, 0);
+				break;
+		}
+
+		vec.X *= SpeedModifier;
+
+		vec.X = Mathf.IsEqualApprox(vec.X, 0f, joyDeadZone * SpeedModifier) ? 0 : vec.X;
```

Comment:
It doesn't. Try setting 50% deadzone and 50% speedlimit and pulling the trigger halfway. On this branch (because of line 54) velocity will be set to 0.25+- and when pulling harder it will smoothly rise to 0.50. On master velocity changes only when the trigger fully pulled. 0 to .5 instantly. If you were to leave deadzone to 0.15 (default) and set speedlimiter to .1 (for kids and stuff) nothing would happen on master even if the trigger is fully pulled (unlike here where it's smooth 0.015 to 0.1). Without this change deadzone becomes basicly "the smallest acceptable velocity value".

Reply:
I have nothing to add.